sqldb/sqlc/queries/invoices.sql

```diff
@@ -19,42 +19,49 @@ SELECT *
 FROM invoice_features
 WHERE invoice_id = $1;
 
--- name: DeleteInvoiceFeatures :exec
-DELETE 
-FROM invoice_features
-WHERE invoice_id = $1;
-
 -- This method may return more than one invoice if filter using multiple fields
 -- from different invoices. It is the caller's responsibility to ensure that 
 -- we bubble up an error in those cases.
 
 -- name: GetInvoice :many
-SELECT *
-FROM invoices
+SELECT i.*
+FROM invoices i
+LEFT JOIN amp_invoices a on i.id = a.invoice_id
 WHERE (
-    id = sqlc.narg('add_index') OR 
+    i.id = sqlc.narg('add_index') OR 
     sqlc.narg('add_index') IS NULL
 ) AND (
-    hash = sqlc.narg('hash') OR 
+    i.hash = sqlc.narg('hash') OR 
     sqlc.narg('hash') IS NULL
 ) AND (
-    preimage = sqlc.narg('preimage') OR 
+    i.preimage = sqlc.narg('preimage') OR 
     sqlc.narg('preimage') IS NULL
 ) AND (
-    payment_addr = sqlc.narg('payment_addr') OR 
+    i.payment_addr = sqlc.narg('payment_addr') OR 
     sqlc.narg('payment_addr') IS NULL
+) AND (
+    a.set_id = sqlc.narg('set_id') OR
```

Comment:
So the AMP invoice is just the singleton instance with the `payment_addr`. A `set_id` value denotes a payment.